cleanlab/segmentation/summary.py

```diff
@@ -227,37 +226,47 @@ def common_label_issues(
       where each row contains information about a particular given/predicted label swap.
       Rows are ordered by the number of label issues inferred to exhibit this type of label swap.
     """
-    try:
-        N, K, H, W = pred_probs.shape
-    except:
+    pred_probs_shape = pred_probs.shape
+    if len(pred_probs_shape) != 4:
         raise ValueError("pred_probs must be of shape (N, K, H, W)")
+    N, K, H, W = pred_probs_shape
 
     assert labels.shape == (N, H, W), "labels must be of shape (N, H, W)"
 
     class_names, exclude, top = _get_summary_optional_params(class_names, exclude, top)
     # Find issues by pixel coordinates
-    issue_coords = np.column_stack(np.where(issues))
+    issue_coords = np.where(issues)
+
+    label_coord_issues = labels[issue_coords]
+    preds = pred_probs[issue_coords[0], :, issue_coords[1], issue_coords[2]].argmax(axis=1)
 
+    mask = ~np.isin(preds, exclude)
+    unique_labels = np.unique(labels)
+
+    if verbose:
+        from tqdm.auto import tqdm
+
+        pbar = tqdm(desc="Labels processed", total=len(unique_labels))
     # Count issues per class (given label)
-    count: Dict[int, Any] = {}
-    for i, j, k in tqdm(issue_coords):
-        label = labels[i, j, k]
-        pred = pred_probs[i, :, j, k].argmax()
-        if label not in count:
-            count[label] = np.zeros(K, dtype=int)
-        if pred not in exclude:
-            count[label][pred] += 1
+    count = {label: np.zeros(K, dtype=int) for label in unique_labels}
+    for label in unique_labels:
+        label_mask = mask & (label_coord_issues == label)
+        label_preds, pred_counts = np.unique(preds[label_mask], return_counts=True)
+        for i, pred in enumerate(label_preds):
+            count[label][pred] = pred_counts[i]
```

Comment:
Likely not performance-critical, but
```suggestion
        for (pred, pred_count) in zip(label_preds, pred_counts):
            count[label][pred] = pred_count
```